src/lib/chips/chip.ts

```diff
@@ -22,30 +24,45 @@ export interface MdChipEvent {
 @Component({
   selector: `md-basic-chip, [md-basic-chip], md-chip, [md-chip],
              mat-basic-chip, [mat-basic-chip], mat-chip, [mat-chip]`,
-  template: `<ng-content></ng-content>`,
+  template: `<ng-content></ng-content><div class="md-chip-focus-border"></div>`,
   host: {
     '[class.mat-chip]': 'true',
     'tabindex': '-1',
     'role': 'option',
 
     '[class.mat-chip-selected]': 'selected',
+    '[class.mat-chip-has-remove-icon]': '_hasRemoveIcon',
```

Comment:
Is there any way to avoid having this class on the chip? It seems unnecessarily complicated that the chip has to know whether it contains a remove button.